src/app/portlets/content-types/edit/content-types-edit.component.ts

```diff
@@ -22,120 +17,102 @@ import { DotConfirmationService } from '../../../api/services/dot-confirmation';
-            }
-        });
+    handleFormSubmit(value: any): void {
+        this.isEditMode() ? this.updateContentType(value) : this.createContentType(value);
     }
 
     /**
-     * Combine data from the form and submit to update content types
-     *
-     * @param {any} $event
+     * Check if the component is in edit mode
      *
-     * @memberof ContentTypesEditComponent
+     * @returns {boolean}
+     * @memberof ContentTypesFormComponent
      */
-    public handleFormSubmit($event): void {
-        const contentTypeData: ContentType = Object.assign({}, this.data, $event.value);
-        this.crudService
-            .putData(`v1/contenttype/id/${this.data.id}`, contentTypeData)
-            .subscribe(this.handleFormSubmissionResponse.bind(this));
+    isEditMode(): boolean {
+        return !!(this.data && this.data.id);
+    }
+
+    /**
+     * Remove fields from the content type
+     * @param fieldsToDelete Fields to be removed
+     */
+    removeFields(fieldsToDelete: Field[]): void {
+        this.fieldService
+            .deleteFields(this.data.id, fieldsToDelete)
+            .pluck('fields')
+            .subscribe((fields: Field[]) => {
+                this.fields = fields;
+            });
     }
 
     /**
-     * Save fields
+     * Save fields to the content type
      * @param fieldsToSave Fields to be save
      */
     saveFields(fieldsToSave: Field[]): void {
-        this.fieldService.saveFields(this.data.id, fieldsToSave).subscribe(fields => {
-            this.data.fields = fields;
-            this.form.updateFormFields();
+        this.fieldService.saveFields(this.data.id, fieldsToSave).subscribe((fields: Field[]) => {
+            this.fields = fields;
         });
     }
 
-    /**
-     * Remove fields
-     * @param fieldsToDelete Fields to be removed
-     */
-    removeFields(fieldsToDelete: Field[]): void {
-        this.fieldService.deleteFields(this.data.id, fieldsToDelete)
-            .pluck('fields')
-            .subscribe(fields => {
-                this.data.fields = <Field[]> fields;
-                this.form.updateFormFields();
+    private createContentType(value: any): void {
+        const data = this.getCreateContentTypeData(value);
+
+        this.crudService
+            .postData('v1/contenttype', data)
+            .flatMap((contentTypes: ContentType[]) => contentTypes)
+            .take(1)
+            .subscribe((contentType: ContentType) => {
+                this.data = contentType;
+                this.fields = this.data.fields;
+                this.form.resetForm();
+                this.location.replaceState(`/content-types-angular/edit/${this.data.id}`);
             });
     }
 
-    private handleFormSubmissionResponse(res: any): void {
-        this.form.resetForm();
+    private getCreateContentTypeData(value: any): ContentType {
+        return Object.assign({}, CONTENT_TYPE_INITIAL_DATA, value);
```

Comment:
is this really necessary? isn't this do by the resolver?

Reply:
I change this, now the form control handle ALL the Content Type object even when some properties are not represent for actual fields in the HTML